### PR TITLE
Adding initial version of AppEntitlements api

### PR DESCRIPTION
**Pull request by Bimych**
Have added initial version of AppEntitlementsApi and tests for that

### CODE REVIEW

**Comment by malaysf** [comment]
@Bimych Can you please work with @pmonks to get the CLA? I will review the code later today.

**Comment by pmonks** [comment]
@malaysf just confirming that @copiesofcopies is working with @Bimych on the CLA side of things.

**Comment by copiesofcopies** [comment]
@malaysf confirming that we now have a valid CLA from @Bimych.

**Comment by malaysf** [comment]
@Bimych There are several APIs related to app entitlement, this change only makes one of theme available. Can you please include the others to make this API complete? As far as I can tell, the remaining ones are:
V1AppEntitlementListPostAsync
V1UserAppEntitlementListGetAsync
V1UserAppEntitlementListPostAsync

**Comment by Bimych** [comment]
Hi @malaysf , sure, will add them all - thanks for pointing out. Additionally would you know why I can't find any V1UserAppEntitlementListGetAsync and V1UserAppEntitlementListPostAsync related documentation anywhere at https://rest-api.symphony.com/ even those code is generated? Do you see their description anywhere?

**Comment by maoo** [comment]
Dear @Bimych , thanks for raising the issue with the cla-bot; I think the problem is due to case sensitivity, I opened an issue on https://github.com/ColinEberhardt/cla-bot/issues/81 . At the same time, I've "fixed" the case-sensitivity of your GitHub ID, therefore, if you push another commit to https://github.com/Bimych/RestApiClient (master branch), the cla-bot should recognise you.

Before moving forward, could you please make sure that your email address is properly configured in your GitHub account? We are aware of an issue of the cla-bot, when the user email is not properly configured, see https://github.com/ColinEberhardt/cla-bot/issues/74

**Comment by malaysf** [comment]
@Bimych Here's the link to the documentation for V1UserAppEntitlementListGetAsync 

https://rest-api.symphony.com/docs/user-apps

I will review the new PR this afternoon.

File: src/SymphonyOSS.RestApiClient/Api/PodApi/AppEntitlementsApi.cs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        public AppEntitlementsApi(IAuthTokens authTokens, string baseUrl, HttpClient httpClient, IApiExecutor apiExecutor)
+        {
+            _log = ApiLogging.LoggerFactory?.CreateLogger<AppEntitlementsApi>();
+
+            _appEntitlementsApi = new AdminClient(baseUrl, httpClient);
+            _authTokens = authTokens;
+            _apiExecutor = apiExecutor;
+        }
+
+        public IEnumerable<PodAppEntitlement> GetAllAppEntitlements()
```
Comment (malaysf) [inline src/SymphonyOSS.RestApiClient/Api/PodApi/AppEntitlementsApi.cs]: Sorry, this comment went unpublished from my previous review.

Can you please create a PodAppEntitlement in the SymphonyOSS.RestApiClient.Api.PodApi namespace? We do not want to expose anything from the generated namespace here to encapsulate all of our implementation details. We have had to change code generators before so keeping all that private gives us future flexibility without resulting in a breaking change for consumers.

In other parts of the project, we have created wrapper functions to convert between the generated classes and the classes that we expose through the public API.

File: src/SymphonyOSS.RestApiClient/Api/PodApi/AppEntitlementsApi.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+            {
+                return _apiExecutor.Execute(_appEntitlementsApi.V1AppEntitlementListGetAsync, _authTokens.SessionToken);
+            }
+            catch (Exception e)
+            {
+                _log?.LogError(0, e, "An error has occured while trying to retrieve application entitlements");
+                throw;
+            }
+        }
+        public IEnumerable<PodAppEntitlement >UpdateAppEntitlements(IEnumerable<PodAppEntitlement> appEntitlements)
```
Comment (malaysf) [inline src/SymphonyOSS.RestApiClient/Api/PodApi/AppEntitlementsApi.cs]: Minor detail but could you please clean up the spacing here:
```
public IEnumerable<PodAppEntitlement >UpdateAppEntitlements
```

to 
```
public IEnumerable<PodAppEntitlement> UpdateAppEntitlements
```

File: src/SymphonyOSS.RestApiClient/Api/PodApi/AppEntitlementsApi.cs (hunk 1 above)
Comment (Bimych) [reply src/SymphonyOSS.RestApiClient/Api/PodApi/AppEntitlementsApi.cs]: Done

File: src/SymphonyOSS.RestApiClient/Api/PodApi/AppEntitlementsApi.cs (hunk 2 above)
Comment (Bimych) [reply src/SymphonyOSS.RestApiClient/Api/PodApi/AppEntitlementsApi.cs]: Done

**Pull request merged**